x/yarpctest/request_unary.go

```diff
@@ -65,11 +66,12 @@ func HTTPRequest(options ...api.RequestOption) api.Action {
 
 // TChannelRequest creates a new tchannel request.
 func TChannelRequest(options ...api.RequestOption) api.Action {
-	opts := api.NewRequestOpts()
-	for _, option := range options {
-		option.ApplyRequest(&opts)
-	}
 	return api.ActionFunc(func(t testing.TB) {
+		opts := api.NewRequestOpts()
```

Comment:
Doesn't moving the `options` processing into the `...Request()` functions make the returned function _not_ idempotent?

eg in this example:
```
myOpts := getOpts()
var x = TChannelRequest(opts)

x.Run() // run with original opts
...

update(myOpts)
x.Run() // run with new opts
```

Since `options` are passed in by reference, we can modify what's being applied to this request action, _after_ this callback is returned.

Reply:
The issue this was trying to solve is that the "opts" object is currently being shared across runs of a single `TChannelRequest`.  If I get an `Action` and attempt to call it twice, we need to have a unique opts struct per request (because the opts defines a few things that are not idempotent).

For the most part, the option pattern should make a copy of the opts unless explicitly passed in.  I would argue that the case you provided should be rare enough that it would be something we could fix in a follow-up.  (And there's always a possibility that people could change something underneath the slice anyway, which screws us again... x folder for now, so #yolo

Reply:
+1 for more yolo